File: 0912-sort-an-array/0912-sort-an-array.java
class Solution {
    public int[] sortArray(int[] nums) {
        
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        for (int num : nums) {
            max = Math.max(max, num);
            min = Math.min(min, num);
        }
        int[] bucket = new int[max - min + 1];
        for (int num : nums) {
            bucket[num - min]++;
        }
        int index = 0;
        for (int i = 0; i < bucket.length; i++) {
            while (bucket[i] > 0) {
                nums[index++] = i + min;
                bucket[i]--;
            }
        }
        return nums;
    }
}

File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        freq_dict = Counter(nums)
                
        min_num, max_num = min(nums),max(nums)
        ans = []
        for num in range(min_num, max_num + 1):
            if num in freq_dict:
                ans.extend([num]*freq_dict[num])
        return ans
        